### ISSUE TITLE
Training and Testing Data Set Exchange per Each Iteration

### ISSUE THREAD

**Issue by zhangpengshan** [ShifuML/shifu#96]
In each iteration, training data set and testing data set is fixed. To make use of all data set, per iteration, re-shuffling training data set and testing data set.

**Comment by huzza** [comment]
Not sure it is a good idea to do in this way. Let's think more. Just have a good sleep:)

**Comment by zhangpengshan** [comment]
Add 'isCrossOver' parameter, if true, exchange training and testing data in each iteration by some probability.  By default it is false.

Configuration example:

 "train" : {
    "baggingNum" : 3,
    ...
     "isCrossOver" : true
 }

Yes, Zhanghao, not sure good performance  for such feature. I am testing and didn't see some improvements. But I guess better model to avoid overfit:)

**Comment by zhangpengshan** [comment]
Testing result is almost the same as not exchanging, why？

**Comment by zhangpengshan** [comment]
Done in develop-0.2.5, may need further testing in 0.2.6.
